docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,307 @@
+
+
+.. _ubuntu-core-image:
+
+Create an image with the real-time kernel
+-----------------------------------------
+
+To do this we need to describe the image content and then use a tool to build it.
+
+.. _model-assertion:
+
+Create the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Inside an empty directory, create a file named ``model.json`` with the above content.
+
+Change the following:
+
+- ``authority-id``, ``brand-id`` to your developer ID, since this is custom model. Use ``snapcraft whoami`` command to get your developer ID.
+- ``timestamp`` to an RFC3339 formatted string that falls within the validity period of your signing key. Use ``date -Iseconds --utc`` command to generate the current time.
+- ``store`` to your dedicated Snap Store ID.
+- ``model`` to a name that accurately represents your device(s).
+
+The ``snaps`` array is a list of snaps that get included in the image.
+In that list, the ``realtime-kernel`` snap contains the realtime Linux kernel.
+Here you can add any other snaps, including for example your real-time applications.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+Use ``snapcraft list-keys`` to check your existing keys.
+If you don't already have a key, create one locally and register it with your account.
+Choose a meaningful name for the key, as it can be used to sign multiple models.
+In this guide we use ``rtu-model`` to sign all our real-time Ubuntu models.
+
+.. code-block:: shell
+
+    snapcraft create-key rtu-model
+    snapcraft register-key rtu-model
+
+Remember to update the model assertion's ``timestamp``, if you created a new key and plan to use it next.
+
+2) Sign the model assertion
+
+.. code-block:: shell
+
+    snap sign -k rtu-model model.json > model.signed.yaml
+
+The ``snap sign`` command takes JSON as input and produces YAML as output!
+
+.. tip::
+
+    You need to repeat the signing every time you change the input model, because the signature is calculated based on the model.
+
+
+Build the image
+~~~~~~~~~~~~~~~
+
+First, get familiar with the tooling by referring to the guide on `building Ubuntu Core images`_.
+
+We use ``ubuntu-image`` and need to set the paths to the following as input:
+
+- Exported store credentials
+- Signed model assertion YAML file
+
+Export the store credentials to a file:
+
+.. code-block:: shell
+
+    snapcraft export-login credentials.txt
```

Comment:
Might be worth mentioning that the credentials can be for any user so long as they have the Viewer role in the store specified in the model assertion.

Reply:
Is there an upstream doc in the Core or Snapcraft docs set that describes this? I can't seem to find it.
It would be cleaner to link to the relevant docs instead of explaining the permissions here.

Related question. If the user only has Viewer role, what happens when the run the `snapcraft export-login` command?